test/instrument/control/if.t

```diff
@@ -97,3 +97,116 @@ tail position iff the whole if-expression is in tail position.
     else (
       ___bisect_visit___ 4;
       print_endline "bar")
+
+When the if-then-else is in a sequence, the next-expressions are
+instrumented as expected.
+
+  $ bash ../test.sh <<'EOF'
+  > let _ =
+  >   if bool_of_string "true" then print_endline "foo" else print_endline "bar";
+  >   print_endline "this";
+  >   print_endline "that";
+  > EOF
+  let _ =
+    if bool_of_string "true" then (
+      ___bisect_visit___ 5;
+      ___bisect_post_visit___ 2 (print_endline "foo"))
+    else (
+      ___bisect_visit___ 4;
+      ___bisect_post_visit___ 3 (print_endline "bar"));
+    ___bisect_post_visit___ 1 (print_endline "this");
+    ___bisect_post_visit___ 0 (print_endline "that")
+
+Where there's a raise or a failwith guarded by a conditional, the
+branch itself is instrumented, but the raising expression is not
+post-instrumented...
```

Comment:
What is the incremental value of having this test over the tests in `apply/special.t`, including https://github.com/aantron/bisect_ppx/blob/e30265643e77bcf2c9eba7322429c779122106fc/test/instrument/apply/special.t#L293 ? Is there any need to test this *in interaction with* `if`-expressions?

Reply:
I don't believe there's a need for this test from the standpoint of the current
implementation. I do see its value in terms of future-proofing and documenting.

The `if cond then raise` pattern is quite common. As it stands, one must be
cautious with how the raising expression is written. It's important to note that
even seemingly innocuous refactoring, which shouldn't affect coverage, could
potentially introduce non-visitable coverage points. This can be quite
surprising.

Consider the following example:

```diff
-| if cond then raise Not_found
+| if cond then raise_s [%sexp "Key not found", { key : Key.t}]
```

This results in the insertion of non-visitable point 1:

```diff
-| if cond then raise Not_found
+| if cond then (__bisect_post__visit 1 (raise_s [%sexp "Key not found", { key : Key.t}]))
``